# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/TextDirectiveUtil.h
# Commit: 09616085d8ef
# Full Hash: 09616085d8ef6470309e1b3c565242d6a811cd5d
# Author: Jan-Niklas Jaeschke <jjaschke@mozilla.com>
# Date: 2025-07-22 09:32:06
# Regressor Bug: 1976473
# File Overlap Count: 5
# Description:
#   Bug 1978235 - Text Fragments: Ensure that start and end terms are always non-empty. r=mccr8,dom-core
#   
#   The patch for Bug 1976473 changed  the behavior of the algorithm to collect word boundary distances
#   to merge punctuation characters with the preceding word. However, this led to unexpected behavior if a
#   context term only consisted of punctuation, where this function would unexpectedly return an empty array.
# ==============================================================================

diff -r df6c5c98b663 -r 09616085d8ef dom/base/TextDirectiveUtil.h
--- a/dom/base/TextDirectiveUtil.h	Mon Jul 21 18:33:07 2025 +0000
+++ b/dom/base/TextDirectiveUtil.h	Mon Jul 21 18:36:11 2025 +0000
@@ -205,6 +205,9 @@
    * used, and the distances are based off the begin of the string.
    * The returned array is always sorted and contains monotonically increasing
    * values.
+   *
+   * This function is guaranteed to return at least one word boundary distance,
+   * the last element always being the length of the string.
    */
   template <TextScanDirection direction>
   static nsTArray<uint32_t> ComputeWordBoundaryDistances(
@@ -532,6 +535,10 @@
                                             ? aString.Length() - wordBegin
                                             : wordEnd);
   }
+  if (wordBoundaryDistances.IsEmpty() ||
+      wordBoundaryDistances.LastElement() != aString.Length()) {
+    wordBoundaryDistances.AppendElement(aString.Length());
+  }
   return std::move(wordBoundaryDistances);
 }
 